contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,711 @@
+    mapping(bytes32 => Claim) public claims; // Stores disputed claims from Vault.
+    mapping(uint256 => uint256) public override externalIDtoLocalID; // Maps external dispute ids to local dispute ids.
+
+    /** @dev Raised when a claim is challenged by claimant.
+     *  @param _claimId Id of the claim in Vault cotract.
+     */
+    event ChallengedByClaimant(bytes32 indexed _claimId);
+
+    /** @dev Raised when a claim is challenged by depositor.
+     *  @param _claimId Id of the claim in Vault cotract.
+     */
+    event ChallengedByDepositor(bytes32 indexed _claimId);
+    
+    /** @dev Raised when a pending claim is validated.
+     *  @param _pendingClaimId Index of the pending claim.
+     */
+    event PendingClaimValidated(uint256 _pendingClaimId);
+
+    modifier onlyGovernor {require(msg.sender == governor, "The caller must be the governor."); _;}
+
+    /** @dev Constructor.
+     *  @param _arbitrator The arbitrator of the contract.
+     *  @param _arbitratorExtraData Extra data for the arbitrator.
+     *  @param _vault Address of the Vault contract.
+     *  @param _metaEvidenceClaimant Metaevidence for the disputes raised by claimant.
+     *  @param _metaEvidenceDepositor Metaevidence for the disputes raised by depositor.
+     *  @param _metaEvidenceSubmit Metaevidence for the disputes that will submit a new claim.
+     *  @param _hackerChallengePeriod Time the hacker has to challenge a claim.
+     *  @param _depositorChallengePeriod Time the depositor has to challenge a claim
+     *  @param _winnerMultiplier Multiplier for calculating the appeal cost of the winning side.
+     *  @param _loserMultiplier Multiplier for calculation the appeal cost of the losing side.
+     *  @param _loserAppealPeriodMultiplier Multiplier for calculating the appeal period for the losing side.
+     */
+    constructor (
+        IArbitrator _arbitrator,
+        bytes memory _arbitratorExtraData,
+        HATVault _vault,
+        string memory _metaEvidenceClaimant,
+        string memory _metaEvidenceDepositor,
+        string memory _metaEvidenceSubmit,
+        uint256 _hackerChallengePeriod,
+        uint256 _depositorChallengePeriod,
+        uint256 _winnerMultiplier,
+        uint256 _loserMultiplier,
+        uint256 _loserAppealPeriodMultiplier
+    ) {
+        emit MetaEvidence(0, _metaEvidenceClaimant);
+        emit MetaEvidence(1, _metaEvidenceDepositor);
+        emit MetaEvidence(2, _metaEvidenceSubmit);
+        
+        governor = msg.sender;
+        arbitrator = _arbitrator;
+        arbitratorExtraData = _arbitratorExtraData;
+        vault = _vault;
+        // Sum of both challenge periods shouldn't exceed challenge period of the vault so the depositor will have time to challenge before the claim is approved in Vault.
+        require(_hackerChallengePeriod + _depositorChallengePeriod <= vault.getChallengePeriod(), "Wrong timeout values");
```

Comment:
The initial spelling is correct. It shouldn't exceed challengeTimeout. otherwise the Vault contract will assume there'll be no challenge and proceed further. This is mostly applies to depositor's challenge in the situation where the hacker's challenge was skipped. In this situation challenge isn't yet flagged by Vault and depositor should be able to have his full `_depositorChallengePeriod `
As for `getChallengeTimeOutPeriod`, it is kind of assumed that this value will be set high enough to contain 2 potential full length disputes

Reply:
yes i see you are of course right